zebra-chain/src/transaction/hash.rs

```diff
@@ -51,7 +66,19 @@ mod tests {
 
         assert_eq!(
             format!("{:?}", hash),
-            "TransactionHash(\"bf46b4b5030752fedac6f884976162bbfb29a9398f104a280b3e34d51b416631\")"
+            r#"TransactionHash("bf46b4b5030752fedac6f884976162bbfb29a9398f104a280b3e34d51b416631")"#
+        );
+    }
+
+    #[test]
+    fn transactionhash_from_str() {
+        let hash = TransactionHash::from_str(
+            "bf46b4b5030752fedac6f884976162bbfb29a9398f104a280b3e34d51b416631",
+        );
```

Comment:
Implementing `FromStr` [allows you to do `str.parse()` rather than `::from_str()`](https://doc.rust-lang.org/stable/rust-by-example/conversion/string.html#parsing-a-string), so we can use that here:

```suggestion
        let hash = "bf46b4b5030752fedac6f884976162bbfb29a9398f104a280b3e34d51b416631".parse();
```